idelib/dataset.py

```diff
@@ -2015,7 +2015,7 @@ def iterMinMeanMax(self, startTime=None, endTime=None, padding=0,
 
     
     def getMinMeanMax(self, startTime=None, endTime=None, padding=0,
-                      times=True, display=False):
+                      times=True, display=False, iterator=iter):
```

Comment:
the parameter `iterator` isn't actually an iterator, so I think the name could stand to be reworked.
- `iterWrapper`? (similar name to built-in `iter` function -> more clearly understood?)
- `onEach`? `onIterEach`?
- `toIterable`?

(this of course applies to all functions using this parameter name scheme)

Reply:
It's not a perfect name, but I don't think it's _entirely_ inappropriate; it's a function/object that iterates another container, which is the definition of the Iterator design pattern. The Python docs defines an iterator as something that implements `__next__()` and raises `StopIteration`, which a generator like I'm using does.

I'll try to come up with a better name, but I've been thinking this might just be left an undocumented argument, intended for internal use, since it may be inherently confusing. Maybe include an advanced example script or something later.

Reply:
I agree with renaming it, something, I'd go for something like onEach or onIter.

I think an easy example could be something like tqdm, basically just adding a command line progress bar.